### PR TITLE
[MM-31404] api4/config: return error when patching PluginSettings.EnableUploads

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Return `403 - Forbidden` if `PuginSettings.EnableUploads` is being patched from REST API, also allow to change it via local API.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-31404

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -1138,6 +1138,10 @@
     "id": "api.config.update_config.clear_siteurl.app_error",
     "translation": "Site URL cannot be cleared."
   },
+  {
+    "id": "api.config.update_config.not_allowed.app_error",
+    "translation": "Changing {{.Name}} is not allowed."
```
Comment (agnivade) [inline i18n/en.json:1143]: Maybe good to add "due to security reasons"

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Review by agnivade — APPROVED** [review]

**Review by lindalumitchell — APPROVED** [review]
Sorry for the delay; old test server setup isn't working. I think this is okay to test post-merge in this case.

**Pull request merged**